submissions/BlueLamantine/friends app/js/main.js

```diff
@@ -0,0 +1,339 @@
+        <input type="radio" id="nameAll" name="sort-name">
+        <label for="nameAll" class="radio-lable">default</label>
+        </div>
+      </fieldset>
+      
+      <fieldset class="filter filter-age">
+        <p>sort by age</p>
+        <div class="radio-wrapper">
+        <input type="radio" id="age09" name="sort-age">
+        <label for="age09" class="radio-lable">young - old</label>
+        <input type="radio" id="age90" name="sort-age">
+        <label for="age90" class="radio-lable">old - young</label>
+        <input type="radio" id="ageAll" name="sort-age">
+        <label for="ageAll" class="radio-lable">default</label>
+        </div>
+      </fieldset>
+
+      
+      <fieldset class="filter filter-search">
+        <p>Search user by email</p>
+        <input id="nameInput" type="text" size="30" class="searchEmail icon search">
+      </fieldset>
+    `;
+  }
+
+  renderFilters() {
+    this.filtersContainer.insertAdjacentHTML('beforeend', this.getHtmlFilter());
+  }
+}
+
+class Filter {
+  constructor(render) {
+    if (Filter.instance == null) {
+      this.profilesContainer = document.querySelector('.profile-cards');
+      this.renderDataWithFilters = render;
+      render.renderFilters();
+      this.filtersPanel = document.querySelector('#filters');
+      this.searchInput = document.querySelector('#nameInput');
+      this.sortedUsers = [];
+      this.sortByNameTypes = {
+        inAscending: 'nameAZ',
+        inDescending: 'nameZA',
+        byDefault: 'nameAll',
+      };
+      this.sortByAgeTypes = {
+        inAscending: 'age09',
+        inDescending: 'age90',
+        byDefault: 'ageAll',
+      };
+      Filter.instance = this;
+    }
+    return Filter.instance;
+  }
+
+  unckeckedRadioButton(nodeName) {
+    const checkedRadio = document.querySelector(
+      `input[type=radio][name=${nodeName}]:checked `
+    );
+    if (checkedRadio) checkedRadio.checked = false;
+  }
+
+  selectSort(selected) {
+    this.searchInput.addEventListener('input', ({ target }) => {
+      let usersByEmail = [];
+      let filter = target.value.toUpperCase();
+
+      this.usersDataForFilter.forEach(user => {
+        if (user.email.toUpperCase().startsWith(filter)) {
+          usersByEmail.push(user);
+        }
+      });
+      this.renderDataWithFilters.usersDataForRender = usersByEmail;
+      this.renderDataWithFilters.renderProfiles();
+    });
+
+    const sort = {
+      'sort-gender': () => {
+        this.unckeckedRadioButton('sort-name');
+        this.unckeckedRadioButton('sort-age');
+        this.sortByGender(selected.id);
+      },
+      'sort-name': () => {
+        this.unckeckedRadioButton('sort-age');
+        this.sortByContent(selected.id, this.sortByNameTypes, 'name');
+      },
+      'sort-age': () => {
+        this.unckeckedRadioButton('sort-name');
+        this.sortByContent(selected.id, this.sortByAgeTypes, 'age');
+      },
+      sortByEmail: () => false,
```

Comment:
I don't see any reason why it should be in this object.

All keys in object named in `kebab-case`, but this method is in `camelCase`.
Also, all keys in your object can be selected by `selected.name` except this one.

Reply:
```sortByEmail: () => false``` removed. All works fine without it